engine/gamesys/src/gamesys/scripts/script_box2d_body.cpp

```diff
@@ -0,0 +1,885 @@
+
+        //{"get_user_data", Body_GetUserData}, - could return the game object id ?
+        //{"set_user_data", Body_SetUserData}, - could attach the body to a game object?
+
+        {"get_mass", Body_GetMass},
+        {"get_inertia", Body_GetInertia},
+        {"get_angle", Body_GetAngle},
+
+        // {"get_mass_data", Body_GetMassData},
+        // {"set_mass_data", Body_SetMassData},
+        // {"reset_mass_data", Body_ResetMassData},
+        // {"synchronize_fixtures", SynchronizeFixtures},
+        // SynchronizeSingle(b2Shape* shape, int32 index)
+
+        {"get_force", Body_GetForce},
+
+        {"get_linear_velocity", Body_GetLinearVelocity},
+        {"set_linear_velocity", Body_SetLinearVelocity},
+
+        {"get_angular_velocity", Body_GetAngularVelocity},
+        {"set_angular_velocity", Body_SetAngularVelocity},
+
+        {"get_linear_damping", Body_GetLinearDamping},
+        {"set_linear_damping", Body_SetLinearDamping},
+
+        {"is_bullet", Body_IsBullet},
+        {"set_bullet", Body_SetBullet},
+
+        {"is_awake", Body_IsAwake},
+        {"set_awake", Body_SetAwake},
+
+        {"is_fixed_rotation", Body_IsFixedRotation},
+        {"set_fixed_rotation", Body_SetFixedRotation},
+
+        {"is_sleeping_allowed", Body_IsSleepingAllowed},
+        {"set_sleeping_allowed", Body_SetSleepingAllowed},
+
+        {"is_active", Body_IsActive},
+        {"set_active", Body_SetActive},
+
+        {"get_gravity_scale", Body_GetGravityScale},
+        {"set_gravity_scale", Body_SetGravityScale},
+
+        {"get_type", Body_GetType},
+        {"set_type", Body_SetType},
+
+        {"get_world_center", Body_GetWorldCenter},
+        {"get_local_center", Body_GetLocalCenter},
+
+        {"get_world_point", Body_GetWorldPoint},
+        {"get_world_vector", Body_GetWorldVector},
+
+        {"get_local_point", Body_GetLocalPoint},
+        {"get_local_vector", Body_GetLocalVector},
+
+        {"get_linear_velocity_from_world_point", Body_GetLinearVelocityFromWorldPoint},
+        {"get_linear_velocity_from_local_point", Body_GetLinearVelocityFromLocalPoint},
+
+        {"apply_force", Body_ApplyForce},
+        {"apply_force_to_center", Body_ApplyForceToCenter},
+        {"apply_torque", Body_ApplyTorque},
+
+        {"apply_linear_impulse", Body_ApplyLinearImpulse},
+        {"apply_angular_impulse", Body_ApplyAngularImpulse},
+
+        {"get_next", Body_GetNext},
+
+        // {"GetFixtureList",GetFixtureList},
+        // {"GetContactList",GetContactList},
+        // {"GetJointList",GetJointList},
+
+        {"get_world", Body_GetWorld},
+        {"dump", Body_Dump},
+
+        {0,0}
+    };
+
+    void ScriptBox2DInitializeBody(struct lua_State* L)
+    {
+        TYPE_HASH_BODY = dmScript::RegisterUserTypeLocal(L, BOX2D_TYPE_NAME_BODY, Body_meta);
+    }
+}
+
+/*# Box2D b2Body documentation
+ *
+ * Functions for interacting with Box2D bodies.
+ *
+ * @document
+ * @name b2body
+ */
+
+/**
```

Comment:
Copied all documentation, but set some to "undocumented" `/**` until they're implemented.